btk/metrics.py

```diff
@@ -1,484 +1,279 @@
+            psnr_results.append(psnr_blend_results)
+            ssim_results.append(ssim_blend_results)
+        else:
+            mse_results.append([-1 for _ in range(len(blend_list[i]))])
+            psnr_results.append([-1 for _ in range(len(blend_list[i]))])
+            ssim_results.append([-1 for _ in range(len(blend_list[i]))])
+    results_reconstruction["mse"] = mse_results
+    results_reconstruction["psnr"] = psnr_results
+    results_reconstruction["ssim"] = ssim_results
+    return results_reconstruction
+
+
+def compute_metrics(
+    blended_images,
+    isolated_images,
+    blend_list,
+    detection_catalogs,
+    segmentations=None,
+    deblended_images=None,
+    use_metrics=("detection", "segmentation", "reconstruction"),
+    meas_band_num=0,
+):
+    results = {}
+    matches = [
+        get_detection_match(blend_list[i], detection_catalogs[i]) for i in range(len(blend_list))
+    ]
+    results["matches"] = matches
+    if "detection" in use_metrics:
+        results["detection"] = detection_metrics(
+            blended_images, isolated_images, blend_list, detection_catalogs, matches
+        )
+    if "segmentation" in use_metrics:
+        results["segmentation"] = segmentation_metrics(
+            blended_images,
+            isolated_images,
+            blend_list,
+            detection_catalogs,
+            segmentations,
+            matches,
+            meas_band_num,
+        )
+    if "reconstruction" in use_metrics:
+        results["reconstruction"] = reconstruction_metrics(
+            blended_images,
+            isolated_images,
+            blend_list,
+            detection_catalogs,
+            deblended_images,
+            matches,
+        )
+    names = blend_list[0].colnames
+    names += ["detected", "distance_closest"]
+    if "reconstruction" in use_metrics:
+        names += ["mse", "psnr", "ssim"]
+    if "segmentation" in use_metrics:
+        names += ["iou"]
+    results["galaxy_summary"] = astropy.table.Table(names=names)
+    for i, blend in enumerate(blend_list):
+        for j, gal in enumerate(blend):
+            row = astropy.table.Table(gal)
+            row.add_column([False], name="detected")
+            row["detected"] = matches[i]["match_detected_id"][j] != -1
+            row.add_column([0.0], name="distance_closest")
+            row["distance_closest"] = matches[i]["dist"][j]
+            if "reconstruction" in use_metrics:
+                row.add_columns([[0.0], [0.0], [0.0]], names=["mse", "psnr", "ssim"])
+                row["mse"] = results["reconstruction"]["mse"][i][j]
+                row["psnr"] = results["reconstruction"]["psnr"][i][j]
+                row["ssim"] = results["reconstruction"]["ssim"][i][j]
+            if "segmentation" in use_metrics:
+                row.add_column([0.0], name="iou")
+                row["iou"] = results["segmentation"]["iou"][i][j]
+            results["galaxy_summary"].add_row(row[0])
```

Comment:
Thanks @thuiop I liked what you did here and you are right, I think we can organize all the information into a single table that will contain all the information we need for any plots/summaries users might be interested in producing. 

Correct me if I'm wrong, but I think you combined all the blends in a batch into a single table right? Maybe we can add a column called `blend_id` to keep track of which galaxies where in which blend?

Reply:
I added the blend_id